### PR TITLE
feat: updated LSJ entries

### LINKED ISSUE #277: Migrate to new LSJ datasource
**Is your feature request related to a problem? Please describe.**
Our current datasource for the LSJ is protected by a BSD license

**Describe the solution you'd like**
We should rebuild the import to use [this datasource](https://raw.githubusercontent.com/STEPBible/STEPBible-Data/master/Lexicons/TFLSJ%20extra%20-%20Translators%20Formatted%20full%20LSJ%20Bible%20lexicon%20-%20STEPBible.org%20CC%20BY.txt) instead.

**Additional context**
We will also need to sort out which words come from Bill Mounce's lexicon and either filter them out, or get permission to use.
We should also credit step bible in our project readme and potentially in the app as well

### PR DESCRIPTION
**Pull request by Pertempto**
## What has changed

- Added data source section to `README`
- Updated greek data file
- Updated LSJ import logic to support the new data file
- Modified BDB import logic
- Updated seed file

## Connected Issues

Closes #277 

## QA Steps

- `nx run db:import-lsj`
- In database: `SELECT * FROM "LemmaResource" WHERE "resourceCode" = 'LSJ';`
    - NOTE: the definitions are multi-line. Click on one to see the whole entry.

## Post-Deployment

- [ ] Run `nx run db:import-db` for production DB

### CODE REVIEW

**Comment by Pertempto** [comment]
@arrocke What did you mean by 
> We will also need to sort out which words come from Bill Mounce's lexicon and either filter them out, or get permission to use.

I didn't do anything with that

File: data/seed.dump
Comment (Pertempto) [inline data/seed.dump:1]: It's strange that this file went down so much in size. I reset my DB to the seed file, and the entries are included.

**Comment by arrocke** [comment]
> We will also need to sort out which words come from Bill Mounce's lexicon and either filter them out, or get permission to use.

I couldn't find a way to identify which entries are from his lexicon. Let's ignore for now

**Review by arrocke — APPROVED** [review]

File: packages/db/src/scripts/helpers/parse-lexicon.ts [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
     }
+  });
+  return parsed;
+};
+
+export const parseLsj = async (filename: string) => {
+  const parsed: Record<string, string> = {};
+  await lineByLine(filename, (line) => {
+    const split = line.split('\t');
+    const id = split[1].split('=')[0].trim();
```
Comment (arrocke) [inline packages/db/src/scripts/helpers/parse-lexicon.ts]: I think we are going to be using the eStrongs number across the board. In Hebrew that aligns with the BDB. #248 should give us a datasource that aligns with eStrongs so we'll be able to update each word to match these LSJ entries. Can you modify this to use that as the strongs?

**Review by arrocke — CHANGES_REQUESTED** [review]
Before this get's merged, let's merge #288 and then rebuild this seed with the LSJ data.

File: packages/db/src/scripts/helpers/parse-lexicon.ts (hunk 1 above)
Comment (Pertempto) [reply packages/db/src/scripts/helpers/parse-lexicon.ts]: Then how should I handle multiple entries for the same eStrongs number?

image

Comment (Pertempto) [reply packages/db/src/scripts/helpers/parse-lexicon.ts]: I was using the `dStrong` column so that we could differentiate between these entries.

Comment (arrocke) [reply packages/db/src/scripts/helpers/parse-lexicon.ts]: For now, let's take the first entry. dStrongs come from StepBible and are used to separate spellings and submeanings of words, and it's not clear to me whether that makes sense for us. I'm still working with Andrew to wrap my head around all of this

Comment (Pertempto) [reply packages/db/src/scripts/helpers/parse-lexicon.ts]: Done. Also improved parsing of bold/italics indicators.

**Review by arrocke — APPROVED** [review]

**Pull request merged**